### PR TITLE
Remove dropDownButton's list items selection if the widget does not use selection mode (T838962)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

**Review by DokaRus — DISMISSED** [review]

File: js/ui/drop_down_button.js [hunk 1]
```diff
@@ -635,6 +641,22 @@ let DropDownButton = Widget.inherit({
         this._popup && this._popup.repaint();
     },
 
+    _selectModeChanged(value) {
+        if(this._list) {
+            this._setListOption('selectionMode', value ? 'single' : 'none');
+            if(value) {
```
Comment (San4es) [inline js/ui/drop_down_button.js]: What about value = 0 case?

File: js/ui/drop_down_button.js [hunk 2]
```diff
@@ -635,6 +641,22 @@ let DropDownButton = Widget.inherit({
...
+                const selectedItemKey = this.option('selectedItemKey');
+                this._setListOption('selectedItemKeys', selectedItemKey ? [selectedItemKey] : []);
+            }
+        }
+        if(!value) {
```
Comment (San4es) [inline js/ui/drop_down_button.js]: value = 0?

File: js/ui/drop_down_button.js (hunk 1 above)
Comment (alexander-kotov-dx) [reply js/ui/drop_down_button.js]: It is the useSelectMode option value and has a type: "boolean".

Comment (San4es) [reply js/ui/drop_down_button.js]: Got it.

File: js/ui/drop_down_button.js [hunk 3]
```diff
@@ -635,6 +641,22 @@ let DropDownButton = Widget.inherit({
         this._popup && this._popup.repaint();
     },
 
+    _selectModeChanged(value) {
+        if(this._list) {
```
Comment (San4es) [inline js/ui/drop_down_button.js]:
```suggestion         
        if(value) {
            this._setListOption('selectionMode', 'single');
            const selectedItemKey = this.option('selectedItemKey');
            this._setListOption('selectedItemKeys', selectedItemKey ? [selectedItemKey] : []);
        } else {
            this._setListOption('selectionMode', 'none');
            this.option({
                'selectedItemKey': undefined,
                'selectedItem': undefined
             });
        }
```

**Review by San4es — DISMISSED** [review]
LGTM

**Review by DokaRus — APPROVED** [review]

**Pull request merged**